src/Microsoft.IdentityModel.Tokens/TokenUtilities.cs

```diff
@@ -136,6 +136,10 @@ internal class TokenUtilities
                             jsonClaimValue
                         }; 
                     }
+                    else if (existingValue is IList<object> existingList)
```

Comment:
You're basically doing the same check twice 😅 

Something like that might make more sense:

```csharp
if (existingValue is not IList<object> existingList)
{
    payload[claim.Type] = new List<object>
    {
        existingValue,
        jsonClaimValue
    }; 
}
else
{
    existingList.Add(jsonClaimValue);
}
```

(you can also invert the two branches to avoid the `not`)

Reply:
fair point 😆 

I'll clean this up.